src/neuromaps_nhp/graph.py

```diff
@@ -0,0 +1,394 @@
+
+        self.graph.add_edge(
+            transform.source_space,
+            transform.target_space,
+            key=edge_key,  # Unique key for this specific transform
+            transform=transform,
+            density=transform.density,
+            hemisphere=transform.hemisphere,
+            file_path=transform.file_path,
+        )
+
+    def get_spaces_by_name(self, space_name: str) -> List[Surface_Atlas]:
+        """Get all surface resources for a space."""
+        return self._spaces.get(space_name, [])
+
+    def get_space_surfaces(
+        self,
+        space_name: str,
+        density: Optional[str] = None,
+        hemisphere: Optional[str] = None,
+    ) -> List[Surface_Atlas]:
+        """Get filtered surface resources for a space."""
+        surfaces = self.get_spaces_by_name(space_name)
+
+        if density:
+            surfaces = [s for s in surfaces if s.density == density]
+        if hemisphere:
+            surfaces = [s for s in surfaces if s.hemisphere == hemisphere]
+
+        return surfaces
+
+    def get_transforms(
+        self,
+        source: str,
+        target: str,
+        density: Optional[str] = None,
+        hemisphere: Optional[str] = None,
+        resource_type: Optional[str] = None,
+    ) -> List[Surface_Transform]:
+        """Get transforms between two spaces with optional filtering."""
+        transforms: List[Surface_Transform] = []
+
+        if self.graph.has_edge(source, target):
+            # Get all edges between source and target
+            edge_data = self.graph.get_edge_data(source, target)
+
+            for key, data in edge_data.items():
+                transform = data["transform"]
+
+                # Apply filters
+                if density and transform.density != density:
+                    continue
+                if hemisphere and transform.hemisphere != hemisphere:
+                    continue
+                if resource_type and transform.resource_type != resource_type:
+                    continue
+                transforms.append(transform)
+
+        return transforms
+
+    def get_all_transforms_for_space(
+        self, space_name: str
+    ) -> Dict[str, List[Surface_Transform]]:
+        """Get all transforms involving a space (incoming and outgoing)."""
+        transforms: Dict[str, List[Surface_Transform]] = {
+            "outgoing": [],
+            "incoming": [],
+        }
+
+        # Outgoing transforms
+        for target in self.graph.successors(space_name):
+            edge_data = self.graph.get_edge_data(space_name, target)
+            for key, data in edge_data.items():
+                transforms["outgoing"].append(data["transform"])
+
+        # Incoming transforms
+        for source in self.graph.predecessors(space_name):
+            edge_data = self.graph.get_edge_data(source, space_name)
+            for key, data in edge_data.items():
+                transforms["incoming"].append(data["transform"])
```

Comment:
I don't believe we are planning on having the incoming transforms, just the outgoing? Will this be used anywhere / will things break if removed?